pkgs/by-name/al/alist/update.sh

```diff
@@ -0,0 +1,27 @@
+#! /usr/bin/env nix-shell
+#! nix-shell -i bash -p gnused coreutils curl jq common-updater-scripts
+
+set -euo pipefail
+
+# get old info
+nixFile=$(nix-instantiate --eval --strict -A "alist.meta.position" | sed -re 's/^"(.*):[0-9]+"$/\1/')
+oldVersion=$(nix-instantiate --eval --strict -A "alist.version" | jq -e -r)
+
+get_latest_release() {
+    local repo=$1
+    curl --fail ${GITHUB_TOKEN:+ -H "Authorization: bearer $GITHUB_TOKEN"} \
+         -s "https://api.github.com/repos/AlistGo/$repo/releases/latest" | jq -r ".tag_name"
+}
+
+version=$(get_latest_release "alist")
+version="${version#v}"
+webVersion=$(get_latest_release "alist-web")
+
+if [[ "$oldVersion" == "$version" ]]; then
+    echo "Already up to date!"
+    exit 0
+fi
+
+update-source-version alist $webVersion --source-key=web --version-key=webVersion
+
+update-source-version alist $version
```

Comment:
I do not think `update-source-version` updates `vendorHash`.  IIRC, only `nix-update-script` supports updating `vendorHash`.

Reply:
Yes, `update-source-version` will not update vendorHash, but using `nix-update` will cause `webVersion` to be overwritten. The solution is to run `update-source-version` for web after `nix-update` to ensure that `webVersion` is correct.

Reply:
> using `nix-update` will cause `webVersion` to be overwritten. The solution is to run `update-source-version` for web after `nix-update` to ensure that `webVersion` is correct.

Please add a comment about this.